src/GraphType.jl

```diff
@@ -275,13 +289,25 @@ mutable struct Graph
                 get!(req, pdict[uuid]) do; VersionSpec() end
             end
             # Translate the requirements into bit masks
-            req_msk = Dict(p1 => (pvers[p1][1:(spp[p1]-1)] .∈ vs) for (p1,vs) in req)
+            # Hot code
+            req_msk = Dict{Int,BitVector}()
+            @inbounds for (p1, vs) in req
+                pv = pvers[p1]
+                req_msk_p1 = BitArray(spp[p1] - 1)
+                for i in 1:spp[p1] - 1
+                    req_msk_p1[i] = pv[i] ∈ vs
+                end
+                req_msk[p1] = req_msk_p1
+            end
             extended_deps[p0][v0] = req_msk
         end
 
         gadj = [Int[] for p0 = 1:np]
         gmsk = [BitMatrix[] for p0 = 1:np]
-        gconstr = [trues(spp[p0]) for p0 = 1:np]
+        local gconstr
+        let spp = spp
+            gconstr = [trues(spp[p0]) for p0 = 1:np]
+        end
```

Comment:
Same here, what's the `let` block for?

Reply:
https://github.com/JuliaLang/julia/issues/15276. Otherwise `spp` gets "tainted" and promoted to a `Core.Box` due to its usage in a closure (comprehension).

Reply:
See e.g. this comment: https://github.com/JuliaLang/julia/issues/15276#issuecomment-297596373